src/Microsoft.ML.TimeSeries/SRCNNAnomalyDetector.cs

```diff
@@ -0,0 +1,269 @@
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        IStatefulTransformer IStatefulTransformer.Clone()
+        {
+            var clone = (SrCnnAnomalyDetector)MemberwiseClone();
+            clone.InternalTransform.StateRef = (SrCnnAnomalyDetectionBaseCore.State)clone.InternalTransform.StateRef.Clone();
+            clone.InternalTransform.StateRef.InitState(clone.InternalTransform, InternalTransform.Host);
+            return clone;
+        }
+
+        internal SrCnnAnomalyDetector(IHostEnvironment env, Options options)
+            : base(new SrCnnArgument(options), LoaderSignature, env)
+        {
+        }
+
+        internal SrCnnAnomalyDetector(IHostEnvironment env, ModelLoadContext ctx)
+            : base(env, ctx, LoaderSignature)
+        {
+        }
+
+        private SrCnnAnomalyDetector(IHostEnvironment env, SrCnnAnomalyDetector transform)
+           : base(new SrCnnArgument(transform), LoaderSignature, env)
+        {
+        }
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            InternalTransform.Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // <base>
+            base.SaveModel(ctx);
+        }
+    }
+
+    /// <summary>
+    /// Detect anomalies in time series using Spectral Residual(SR) algorithm
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///To create this estimator, use [DetectAnomalyBySrCnn](xref:Microsoft.ML.TimeSeriesCatalog.DetectAnomalyBySrCnn(Microsoft.ML.TransformsCatalog,System.String,System.String,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.Double))
+    ///###  Estimator Characteristics
+    ///|  |  |
+    ///| -- | -- |
+    ///| Does this estimator need to look at the data to train its parameters? | No |
+    ///| Input column data type | <xref:System.Single> |
+    ///| Output column data type | 3-element vector of<xref:System.Double> |
+    ///### Background
+    ///At Microsoft, we develop a time-series anomaly detection service
+    ///which helps customers to monitor the time-series continuously
+    ///and alert for potential incidents on time. To tackle the problem
+    ///of time-series anomaly detection, we propose a novel algorithm
+    ///based on Spectral Residual (SR) and Convolutional Neural Network
+    ///(CNN). The SR model is borrowed from visual saliency detection domain to time-series anomaly detection. And here we onboarded this SR algorithm firstly.
+    ///
+    ///The Spectral Residual (SR) algorithm is unsupervised, which means training step is not needed while using SR. It consists of three major steps:
+    ///(1) Fourier Transform to get the log amplitude spectrum;
+    ///(2) calculation of spectral residual;
+    ///(3) Inverse Fourier Transform that transforms the sequence back to spatial domain.
```

Comment:
Please add a space between `///` and text.

Reply:
Done.